node/node.go

```diff
@@ -240,5 +242,21 @@ func (n *Node) PurgeAndRestart(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	if n.acp.HasValue() {
+		acp := n.acp.Value()
+		err := acp.DropAll(ctx)
+		if err != nil {
+			// for now we will just log this error, since SourceHub ACP doesn't yet
+			// implement the DropAll.
+			log.ErrorE("Failed to drop ACP state", err)
+		}
+		// follow up close call on ACP is required since the node.Start function starts
+		// ACP again anyways so we need to gracefully close before starting again
+		err = acp.Close()
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
question: Is this why you needed to introduce the `acp immutable.Option[acp.ACP]` variable on the `Node struct`?

todo: If so, why not just operate on the acp instance that is already on `n.DB` (just introduce a method on the `client.DB` interface like the existing other acp methods. OR just add this logic to `n.DB.Close()` which is called by `Node.Close()` which is called in the function above anyway.

Reply:
Happy with reasoning will change my todo to non-blocking. Giving lgtm